test/core_spec.js

```diff
@@ -220,4 +220,53 @@ describe('application logic', () => {
       }));
     });
   });
+
+  describe('addMessage', () => {
+    it('adds a message to chat with user and timestamp', () => {
+      let user = "cat";
+      let message = "meow";
+
+      let chat = Game.addMessage(List.of(Map({
+          timestamp: Date.now(), user: 'Alice', message: 'hi'
+        }), Map({
+          timestamp: Date.now(), user: 'Bob', message: 'hi'
+        })), user, message);
+      expect(chat).to.exist;
+      expect(chat.has(2)).to.be.true;
+
+      let msg = chat.get(2);
+      expect(msg.has('timestamp')).to.be.true;
+      expect(msg.get('timestamp')).to.be.a('number');
+      expect(msg.get('user')).to.equal(user);
+      expect(msg.get('message')).to.equal(message);
+    });
+
+    it('truncates long usernames', () => {
+      let user = "hi-i-have-a-long-username";
+      let message = "meow";
+
+      let chat = Game.addMessage(List(), user, message);
+      expect(chat).to.exist;
+
+      let msg = chat.first();
+      expect(msg.has('timestamp')).to.be.true;
+      expect(msg.get('timestamp')).to.be.a('number');
+      expect(msg.get('user')).to.equal(user.substr(0, 20));
+      expect(msg.get('message')).to.equal(message);
+    });
+
+    it('truncates long messages', () => {
+      let user = "cat";
+      let message = "qewrieutowuqpoeiurpoqweiurpoqiweuroidjaldkxcnzjxkviushfnuawhepurnhawufncdisocjozixdcoijnwioenjaiodnqewrieutowuqpoeiurpoqweiurpoqiweuroidjaldkxcnzjxkviushfnuawhepurnhawufncdisocjozixdcoijnwioenjaiodnas";
```

Comment:
this overflows line limit.

"a".repeat(250) probably serves fine for the purposes of this test